controllers/kubevirtmachine_controller.go

```diff
@@ -339,6 +341,17 @@ func (r *KubevirtMachineReconciler) reconcileNormal(ctx *context.MachineContext)
 		ctx.KubevirtMachine.Status.Ready = false
 	}
 
+	// If the kubevirtMachine's VM is not managed using the default infra client
+	// provided to this pod, then we are not watching the object using an informer.
+	// Use the VM poll interval with some jitter to ensure the reconcile for this VM
+	// wakes up periodically. This is important to detect state changes such as an
+	// IP change
+	if ctx.KubevirtMachine.Spec.InfraClusterSecretRef != nil {
+		// add 10% jitter to the requeue period
+		requeueAfter := r.ExternalClusterSyncPeriod + time.Duration(rand.Intn(int(r.ExternalClusterSyncPeriod)/10))
```

Comment:
There is a Jitter function you can use for this https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#Jitter

Reply:
updated